README
NAME
    Dist::Zilla::Plugin::MetaProvides::Class - Scans Dist::Zilla's .pm files
    and tries to identify classes using Class::Discover.

VERSION
    version 2.001001

SYNOPSIS
      [MetaProvides::Class]
      meta_noindex    = 1  ; default > See :MetaProvider::Provider
      inherit_version = 1  ; default > See :MetaProvider::Provider
      inherit_missing = 1  ; default > See :MetaProvider::Provider

ROLES
  "::MetaProvider::Provider"
    "→ Dist::Zilla::Role::MetaProvider::Provider"

   "meta_noindex"
    Extended from "MetaProvider::Provider"

    This is a utility for people who are also using "MetaNoIndex", so that
    its settings can be used to eliminate items from the 'provides' list.

    *   meta_noindex = 0

        By default, do nothing unusual.

    *   DEFAULT: meta_noindex = 1

        When a module meets the criteria provided to "MetaNoIndex",
        eliminate it from the metadata shipped to "Dist::Zilla"

ROLE SATISFYING METHODS
  provides
    A conformant function to the "::MetaProvider::Provider" Role.

   signature: $plugin->provides()
   returns: Array of ":MetaProvides::ProvideRecord"
PRIVATE METHODS
  _classes_for
   signature: $plugin->_classes_for( $filename, $file_content )
   returns: Array of ":MetaProvides::ProvideRecord"
SEE ALSO
    *   "[MetaProvides]"

AUTHOR
    Kent Fredric <kentnl@cpan.org>

COPYRIGHT AND LICENSE
    This software is copyright (c) 2014 by Kent Fredric
    <kentfredric@gmail.com>.

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.



Changes
Revision history for Dist-Zilla-Plugin-MetaProvides-Class

2.001002 2017-03-08T10:11:02Z 7e3ff35

2.001001 2017-03-08T10:08:02Z 3334cdb
 - Bugfix: Avoid test failures with -Ddefault_inc_excludes_dot
 - Remove usage of Test::KENTNL/dztest

 [Dependencies::Stats]
 - Dependencies changed since 2.001000, see misc/*.deps* for details
 - configure: +1 (recommends: ↑1)
 - develop: +11 ↑3 -3 (recommends: +1, suggests: ↑2)
 - runtime: ↓1 -1 (recommends: +1)
 - test: +1 -1 (recommends: ↑1 ↓1)

2.001000 2014-10-16T19:13:57Z 7d672f0
 [Dependencies::Stats]
 - Dependencies changed since 2.000000, see misc/*.deps* for details
 - configure: ↓1 (recommends: ↑1)
 - develop: +7 ↑2 -3 (suggests: +2 -1)
 - runtime: -1
 - test: +2 ↑1 -4 (recommends: ↑3 -1)

 [Internals]
 - Stop Autoboxing

 [Metadata]
 - move repo to kentnl org
 - add copyright holder details

 [Tests]
 - Internals reimplemented with newer dztest tool

2.000000 2014-05-01T12:15:40Z
 [00 Minor]
 - No substantial code changes in this release, mostly packaging.
 - Mostly hands free except for non-cpan downstream vendors who might work by HAND

 [01 Toolkit Change]
 - EUMM is now used instead of Module::Build
 - EUMM/Test depedencies are now only escallated to latest as recommendations.

 [02 Version Scheme Change]
 - X.12345678 is now X.123456 ( mantissa 8 -> 6 )
 - This is the primary reason for the 2.x on the box.

 [Dependencies::Aggregate]
 - 5.008009 : 217
 - 5.010001 : 196
 - 5.018002 : 181
 - 5.019009 : 177

 [Dependencies::Stats]
 - Dependencies changed since 1.14000001, see misc/*.deps* for details
 - build: -1
 - configure: +1 -1 (recommends: +1)
 - develop: +43 -2 (recommends: -1, suggests: +1 -1)
 - runtime: +1 ↑2
 - test: +4 ↓1 -2 (recommends: +4)

1.14000001 2013-04-26T15:19:42Z
 - Maintenance Release

 [Dependencies::Noteworthy]
 - Module::Build 0.3601 → 0.4004
 - Test::More 0.96 → 0.98

 [Dependencies::Stats]
 - Dependencies changed since 1.14000000, see misc/*.deps* for details
 - build: ↑1
 - configure: ↑1
 - develop: +6 (suggests: ↑1)
 - test: ↑1

 [Documentation]
 - Add README.mkdn
 - Update Copyright Year

 [Meta]
 - Bug Tracker to Github Issues

 [Packaging]
 - utilise test_requires in Module::Build

1.14000000 2012-05-09T07:14:20Z
 - Maintenance release.

 [Dependencies::Noteworthy]
 - Dist::Zilla::MetaProvides::ProvideRecord 0 → 1.14000000

 [Dependencies::Stats]
 - Dependencies changed since 1.12060501, see misc/*.deps* for details
 - develop: (recommends: ↑1, suggests: ↑1)
 - runtime: +3 ↑1
 - test: -1

 [Internals]
 - add $AUTHORITY to all packages
 - move $VERSION out of BEGIN

 [Packaging]
 - Update LICENSE (Year, Indent, Address)
 - Move accessory tests to xt/
 - git uri's moved to https:// versions
 - export x_authority

1.12060501 2011-04-08T13:51:50Z
 [BugFix]
 - rt#67325 : Should now depend on Test::More 0.96 to solve tests failing on earlier versions.

 [Dependencies::Noteworthy]
 - Test::More 0.88 → 0.96

 [Dependencies::Stats]
 - Dependencies changed since 1.12060312, see misc/*.deps* for details
 - test: ↑1

1.12060312 2011-04-07T00:04:18Z
 - Mostly a maintenance release. No user visible changes should exist.

 [Dependencies::Stats]
 - Dependencies changed since 1.12044911, see misc/*.deps* for details
 - develop: +1 -1 (recommends: +1 -1, suggests: +1 -1)

 [Documentation]
 - Fix some bad POD links.

 [Internals]
 - Reorederd some code to reduce cognitive complexity.

 [Packaging]
 - Move to @Author::KENTNL
 - Stricitify critic.
 - Reworked Changes for CPAN::Changes.

 [Tests]
 - Added CPAN::Changes.
 - Removed Portability.

1.12044911 2010-11-07T23:10:31Z
 - Reinsert MooseX::Declare into deps, got lost due to the t/ to corpus/ move.

 [Dependencies::Stats]
 - Dependencies changed since 1.12044910, see misc/*.deps* for details
 - test: +1

1.12044910 2010-11-07T22:56:20Z
 - Update deps to match those of ::Package
 - Add some realistic tests.
 - Bumped the minor. NOTICE: meta_noindex is on by default, so if you use the MetaNoIndex plugin it will do the right
   thing.

 [Dependencies::Noteworthy]
 - suggests: Dist::Zilla::Plugin::MetaNoIndex
 - needs: Dist::Zilla::Util::Test::KENTNL 0.01000011
 - needs: Test::Fatal
 - needs: Test::Moose
 - Dist::Zilla::Plugin::MetaProvides 1.11044405 → 1.12044806

 [Dependencies::Stats]
 - Dependencies changed since 1.11044406, see misc/*.deps* for details
 - develop: +2
 - runtime: ↑1 (suggests: +1)
 - test: +3 -2 (suggests: +1)

1.11044406 2010-11-02T18:03:05Z
 - Update toolchain.
 - Add meta_noindex support from Role::MetaProvides::Provides

 [Dependencies::Noteworthy]
 - Dist::Zilla::Plugin::MetaProvides 1.11034201 → 1.11044405

 [Dependencies::Stats]
 - Dependencies changed since 1.11034201, see misc/*.deps* for details
 - develop: +1 (recommends: +1, suggests: +1)
 - runtime: ↑1
 - test: +1

1.11034201 2010-07-24T13:41:58Z
 - History forks here from Dist::Zilla::Plugin::MetaProvides

1.10034117 2010-07-24T05:37:49Z
 - Dropped xt/author/kwalitee.t to avoid the needless dep. ( Kwalitee Test plugin doesn't have this issue )

1.10027802 2010-05-22T13:08:02Z
 - Replaced most of dist.ini with @KENTNL ( Yeilding more accurate t/ depends )
 - Removed =inc::Dist style loopback in favour of the more reliable and more generic =inc::lib

1.10027518 2010-05-20T05:33:10Z
 - Upgraded tests to Dist::Zilla 2.x. ( should resolve rt #56482 ) ( Thanks to DAXIM/Lars Dɪᴇᴄᴋᴏᴡ 迪拉斯 for reporting )
 - Upgraded dist.ini to Dist::Zilla 2.x

1.10001919 2009-09-12T06:17:57Z
 - Dropped the version dep on FindBin because it causes problems for some. Apparently deping on something that may or
   may not be in core which may or may not be in core later is a bad thing.
 - Added CompileTests
 - Add dist.ini to release for diagnostic reasons.

1.10000417 2009-08-28T04:52:32Z
 - Moved the tests around.
 - More tests.
 - All versions now maxing out to encourage upgrading.
 - Moved to relative dating scheme
 - Tighter DogFeed Loopback in buildsystem.
 - Critic'd.
 - Documentation Upped on the plugs.


t/eg/DZ2/lib/DZ2mx.pm
# ABSTRACT: turns baubles into trinkets
# $Id:$
use strict;
use warnings;

use MooseX::Declare;

class DZ2::Mx {

}

class DZ2::Mk {

}

1;



Makefile.PL
# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v6.009.
use strict;
use warnings;

use 5.006;

use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "Scans Dist::Zilla's .pm files and tries to identify classes using Class::Discover.",
  "AUTHOR" => "Kent Fredric <kentnl\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0
  },
  "DISTNAME" => "Dist-Zilla-Plugin-MetaProvides-Class",
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.006",
  "NAME" => "Dist::Zilla::Plugin::MetaProvides::Class",
  "PREREQ_PM" => {
    "Class::Discover" => 0,
    "Dist::Zilla::MetaProvides::ProvideRecord" => "2.000000",
    "Dist::Zilla::Plugin::MetaProvides" => "1.12044806",
    "Dist::Zilla::Role::MetaProvider::Provider" => 0,
    "Moose" => 0,
    "namespace::autoclean" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "Test::DZil" => 0,
    "Test::Fatal" => 0,
    "Test::Moose" => 0,
    "Test::More" => "0.96"
  },
  "VERSION" => "2.001002",
  "test" => {
    "TESTS" => "t/*.t t/00-compile/*.t"
  }
);


my %FallbackPrereqs = (
  "Class::Discover" => 0,
  "Dist::Zilla::MetaProvides::ProvideRecord" => "2.000000",
  "Dist::Zilla::Plugin::MetaProvides" => "1.12044806",
  "Dist::Zilla::Role::MetaProvider::Provider" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "Moose" => 0,
  "Test::DZil" => 0,
  "Test::Fatal" => 0,
  "Test::Moose" => 0,
  "Test::More" => "0.96",
  "namespace::autoclean" => 0,
  "strict" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);


maint/perlcritic.rc.gen.pl
#!/usr/bin/env perl
## no critic (Modules::RequireVersionVar)

# FILENAME: bundle_to_ini.pl
# CREATED: 02/06/14 01:48:56 by Kent Fredric (kentnl) <kentfredric@gmail.com>
# ABSTRACT: Write an INI file from a bundle

use 5.008;    #utf8
use strict;
use warnings;
use utf8;

use Carp qw( croak carp );
use Perl::Critic::ProfileCompiler::Util qw( create_bundle );

## no critic (ErrorHandling::RequireUseOfExceptions)
my $bundle = create_bundle('Example::Author::KENTNL');
$bundle->configure;

my @stopwords = (
  qw(
    metadata conformant
    )
);
for my $wordlist (@stopwords) {
  $bundle->add_or_append_policy_field( 'Documentation::PodSpelling' => ( 'stop_words' => $wordlist ) );
}

#$bundle->remove_policy('ErrorHandling::RequireCarping');
#$bundle->remove_policy('NamingConventions::Capitalization');

my $inf = $bundle->actionlist->get_inflated;

my $config = $inf->apply_config;

{
  open my $rcfile, '>', './perlcritic.rc' or croak 'Cant open perlcritic.rc';
  $rcfile->print( $config->as_ini, "\n" );
  close $rcfile or croak 'Something fubared closing perlcritic.rc';
}
my $deps = $inf->own_deps;
{
  open my $depsfile, '>', './perlcritic.deps' or croak 'Cant open perlcritic.deps';
  for my $key ( sort keys %{$deps} ) {
    $depsfile->printf( "%s~%s\n", $key, $deps->{$key} );
    *STDERR->printf( "%s => %s\n", $key, $deps->{$key} );
  }
  close $depsfile or carp 'Something fubared closing perlcritic.deps';
}



Build.PL

use strict;
use warnings;

use Module::Build 0.3601;


my %module_build_args = (
  "build_requires" => {
    "Module::Build" => "0.4004"
  },
  "configure_requires" => {
    "Module::Build" => "0.4004"
  },
  "dist_abstract" => "Scans Dist::Zilla's .pm files and tries to identify classes using Class::Discover.",
  "dist_author" => [
    "Kent Fredric <kentnl\@cpan.org>"
  ],
  "dist_name" => "Dist-Zilla-Plugin-MetaProvides-Class",
  "dist_version" => "1.14000001",
  "license" => "perl",
  "module_name" => "Dist::Zilla::Plugin::MetaProvides::Class",
  "recommends" => {},
  "recursive_test_files" => 1,
  "requires" => {
    "Class::Discover" => 0,
    "Dist::Zilla::MetaProvides::ProvideRecord" => "1.14000000",
    "Dist::Zilla::Plugin::MetaProvides" => "1.12044806",
    "Dist::Zilla::Role::MetaProvider::Provider" => 0,
    "Moose" => 0,
    "Moose::Autobox" => 0,
    "namespace::autoclean" => 0,
    "perl" => "5.006",
    "strict" => 0,
    "warnings" => 0
  },
  "script_files" => [],
  "test_requires" => {
    "Dist::Zilla::Util::Test::KENTNL" => "0.01000011",
    "File::Find" => 0,
    "File::Temp" => 0,
    "MooseX::Declare" => 0,
    "Test::Fatal" => 0,
    "Test::Moose" => 0,
    "Test::More" => "0.98"
  }
);


unless ( eval { Module::Build->VERSION(0.4004) } ) {
  my $tr = delete $module_build_args{test_requires};
  my $br = $module_build_args{build_requires};
  for my $mod ( keys %$tr ) {
    if ( exists $br->{$mod} ) {
      $br->{$mod} = $tr->{$mod} if $tr->{$mod} > $br->{$mod};
    }
    else {
      $br->{$mod} = $tr->{$mod};
    }
  }
}

my $build = Module::Build->new(%module_build_args);

$build->create_build_script;
